Xamarin.Forms.Platform.iOS/Forms.cs

```diff
@@ -124,36 +124,63 @@ protected override Expression VisitMember(MemberExpression node)
 			}
 		}
 
-		internal class IOSDeviceInfo : DeviceInfo
+		class IOSDeviceInfo : DeviceInfo
 		{
-			readonly NSObject _notification;
 			readonly Size _scaledScreenSize;
 			readonly double _scalingFactor;
+			bool _disposed;
 
 			public IOSDeviceInfo()
 			{
-				_notification = UIDevice.Notifications.ObserveOrientationDidChange((sender, args) => CurrentOrientation = UIDevice.CurrentDevice.Orientation.ToDeviceOrientation());
-
```

Comment:
We shouldn't be observing device orientation changes for layout changes. Instead, we should observe status bar changes.